json_schema/extensions/x-aep-resource.yaml

```diff
@@ -4,18 +4,24 @@ title: x-aep-resource
 description: Defines the resource name patterns and naming for an AEP API resource
 type: object
 required:
-  - singular
   - plural
+  - parents
   - patterns
+  - singular
+  - type
 properties:
-  singular:
-    type: string
-    description: The singular form of the resource name
-    pattern: "^[a-z][a-z0-9-]*$"
   plural:
     type: string
     description: The plural form of the resource name
     pattern: "^[a-z][a-z0-9-]*$"
+  parents:
+    type: array
+    description: The parent resources of this resource
+    items:
+      type: string
+      pattern: "^[a-z0-9][.-a-z0-9/]*[a-z0-9]$"
+    examples:
+      - "library.example.com/author"
```

Comment:
This does not support upper case? We should double check our docs/examples if so